example/thirdparty/arrow/build.mill

```diff
@@ -0,0 +1,689 @@
+          .head
+      )
+        .filter(_.toIO.isFile)
+      // TODO this is to prevent compilation failure in cases when there is actual declaration in Foo.kt and expect
+      //  declaration in Foo.kt instead of Foo.[jvm,js,...].kt, because in this case symbols belong to the same module.
+      // This is allowed with `-Xcommon-sources` flag, but probably will be prohibited in the future.
+      // Ideally, it should be in the [[KotlinModule]] directly, but this implies [[KotlinModule]] should know where
+      // common sources are, making some assumptions about layout.
+      s"-Xcommon-sources=${files.mkString(",")}"
+    }
+
+    trait ArrowPlatformModule extends KotlinModule {
+      def kotlinVersion = libraries.versions.kotlin
+      def languageVersion = majorVersion(kotlinVersion())
+      def apiVersion = majorVersion(kotlinVersion())
+      def explicitApi = ArrowMultiplatformModule.this.explicitApi
+
+      def platformCrossSuffix: String = millModuleSegments
+        .value
+        .collect { case l: mill.define.Segment.Label => l.value }
+        .last
+
+      override def millSourcePath = super.millSourcePath / os.up
+
+      override def sources: T[Seq[PathRef]] = Task.Sources {
+        val sourcesRootPath = millSourcePath / "src"
+        var sources = Seq("common", platformCrossSuffix)
+          .map(platform => PathRef(sourcesRootPath / s"${platform}Main" / "kotlin"))
+          .filter(p => os.exists(p.path))
+        if (platformCrossSuffix != "jvm") {
+          val nonJvmSourcesPath = sourcesRootPath / "nonJvmMain" / "kotlin"
+          if (os.exists(nonJvmSourcesPath)) {
+            sources ++= Seq(PathRef(nonJvmSourcesPath))
+          }
+        }
+        sources
+      }
+
+      override def kotlincOptions = super.kotlincOptions() ++ Seq("-Xexpect-actual-classes")
+    }
+
+    def jvm: ArrowPlatformJvmModule
+    trait ArrowPlatformJvmModule extends ArrowPlatformModule with KoverModule { outer =>
+      def kotlincOptions = super.kotlincOptions() ++ Seq(commonSourcesArg(sources(), "commonMain"))
+      def test: ArrowPlatformJvmTests
+      trait ArrowPlatformJvmTests extends KotlinTests with KoverTests with TestModule.Junit5 {
+        override def sources: T[Seq[PathRef]] = Task.Sources {
+          val sourcesRootPath = outer.millSourcePath / "src"
+          Seq("common", outer.platformCrossSuffix)
+            .map(platform => PathRef(sourcesRootPath / s"${platform}Test" / "kotlin"))
+            .filter(p => os.exists(p.path))
+        }
+
+        override def compile: T[CompilationResult] =
+          if (!modulesWithTestingEnabled.contains(multiplatformRoot))
+            Task { CompilationResult(T.dest, PathRef(T.dest)) }
+          else super.compile
+        override def test(args: String*): Command[(String, Seq[TestResult])] =
+          if (!modulesWithTestingEnabled.contains(multiplatformRoot))
+            Task.Command { ("", Seq.empty[TestResult]) }
+          else super.test(args: _*)
```

Comment:
These are a bit strange. Can we at least add a comment, what and more important why it is happening here?